tests/e2e/testdata/scenarios/bigquerydataset_unsetfields/script.yaml

```diff
@@ -0,0 +1,102 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: kms.cnrm.cloud.google.com/v1beta1
+kind: KMSKeyRing
+metadata:
+  name: kmskeyring-${uniqueId}
+spec:
+  location: us
+---
+apiVersion: kms.cnrm.cloud.google.com/v1beta1
+kind: KMSCryptoKey
+metadata:
+  labels:
+    key-one: value-one
+  name: kmscryptokey-${uniqueId}
+spec:
+  keyRingRef:
+    name: kmskeyring-${uniqueId}
+---
+apiVersion: serviceusage.cnrm.cloud.google.com/v1beta1
+kind: ServiceIdentity
+metadata:
+  name: serviceidentity-${uniqueId}
+  annotations:
+    cnrm.cloud.google.com/deletion-policy: "abandon"
+spec:
+  projectRef:
+    external: ${projectId}
+  resourceID: bigquery.googleapis.com
+---
+apiVersion: iam.cnrm.cloud.google.com/v1beta1
+kind: IAMPolicy
+metadata:
+  name: iampolicy-${uniqueId}
+spec:
+  resourceRef:
+    apiVersion: kms.cnrm.cloud.google.com/v1beta1
+    kind: KMSCryptoKey
+    name: kmscryptokey-${uniqueId}
+  bindings:
+    - role: roles/cloudkms.cryptoKeyEncrypterDecrypter
+      members:
+        - serviceAccount:bq-${projectNumber}@bigquery-encryption.iam.gserviceaccount.com
+---
+apiVersion: iam.cnrm.cloud.google.com/v1beta1
+kind: IAMServiceAccount
+metadata:
+  annotations:
+    cnrm.cloud.google.com/project-id: "${projectId}"
+  name: iamsa-${uniqueId}
+---
+apiVersion: bigquery.cnrm.cloud.google.com/v1beta1
+kind: BigQueryDataset
+metadata:
+  name: bigquerydataset${uniqueId}
+spec:
+  description: "Fully Configured BigQuery Dataset"
+  friendlyName: bigquerydataset-fullyconfigured
+  defaultPartitionExpirationMs: 3600000
+  defaultTableExpirationMs: 3600000
+  defaultCollation: und:ci
+  defaultEncryptionConfiguration:
+    kmsKeyRef:
+      name: kmscryptokey-${uniqueId}
+  isCaseInsensitive: true
+  location: US
+  maxTimeTravelHours: "72"
+  projectRef:
+    external: ${projectId}
+  access:
+    - role: OWNER
+      specialGroup: projectOwners
+
+---
+
+apiVersion: bigquery.cnrm.cloud.google.com/v1beta1
+kind: BigQueryDataset
+metadata:
+  name: bigquerydataset${uniqueId}
+spec:
+  description: "Fully Configured BigQuery Dataset updated"
+  defaultEncryptionConfiguration:
+    kmsKeyRef:
+      name: kmscryptokey-${uniqueId}
+  location: US
```

Comment:
nit: I suggest that we keep the unset fields in this object but just comment them out. That way it is super clear which fields are unset.